cartography/intel/semgrep/findings.py

```diff
@@ -282,6 +286,12 @@ def sync(
     load_semgrep_sca_vulns(neo4j_sesion, vulns, deployment_id, update_tag)
     load_semgrep_sca_usages(neo4j_sesion, usages, deployment_id, update_tag)
     run_scoped_analysis_job('semgrep_sca_risk_analysis.json', neo4j_sesion, common_job_parameters)
+
+    # fetch and load dependencies for the Go ecosystem
```

Comment:
Now that this `sync()` function is being used for more than just findings, should I move it to a separate file, e.g. common.py? Or I could create a separate `sync()` function in dependencies.py, but it would need to duplicate a lot of this function's code.

Reply:
Good question. I would say an approach like the [github](https://github.com/cartography-cncf/cartography/blob/master/cartography/intel/semgrep/__init__.py) ingestion is good, multiple resource syncs called from the [start_semgrep_ingestion](https://github.com/cartography-cncf/cartography/blob/810e391903bc7271f55e9c6c58d525d2486f6041/cartography/intel/semgrep/__init__.py#L14C5-L14C28).

The only code you need to repeat is the `cleanup` and `merge_module_sync_metadata` process.

Reply:
The other thing is that the findings API requires the deployment ID, currently retrieved here: https://github.com/cartography-cncf/cartography/blob/810e391903bc7271f55e9c6c58d525d2486f6041/cartography/intel/semgrep/findings.py#L275-L278

I could update `start_semgrep_ingestion` to fetch the deployment ID and pass it as an argument to the other sync functions?

Reply:
@heryxpc I've done some refactoring, what do you think? I don't love the way it's set up now but couldn't figure out a cleaner approach, open to suggestions

Reply:
I've taken another pass at organizing the code, this way is much cleaner but it feels a little brittle that `sync_deployment` sets values in `common_job_parameters` that are required by `sync_findings` and `sync_dependencies`, so it would break if you called the sync functions in a different order